services/src/main/java/org/keycloak/protocol/oidc/grants/ciba/endpoints/BackchannelAuthenticationCallbackEndpoint.java

```diff
@@ -81,15 +81,21 @@ public Response processAuthenticationChannelResult(AuthenticationChannelResponse
                     Response.Status.BAD_REQUEST);
         }
 
+        status = preApprove(response);
+        
         switch (status) {
             case SUCCEED:
-                approveRequest(bearerToken, response.getAdditionalParams());
+                approveRequest(bearerToken.getId(), response.getAdditionalParams());
                 break;
             case CANCELLED:
             case UNAUTHORIZED:
-                denyRequest(bearerToken, status);
+                denyRequest(bearerToken.getId(), status);
                 break;
         }
+        
+        if (!postApprove(response)) {
+            denyRequest(bearerToken.getId(), status);
```

Comment:
The logic is hard to understand as `denyRequest()` might have already been called.

Reply:
Actually we don't use this, so I will remove it, as  the only reason it is there is that we possibly could use it. :)